### PR TITLE
[SC-105] Rewrite tests & remove 'aave-helpers' dependency

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: foundry.toml
Comment (hexonaut) [inline foundry.toml]: This should be picked up automatically without needing to manually specify the remapping.

File: .gitmodules
Comment (hexonaut) [inline .gitmodules]: I think we can leave the branch off now as pending your approval that will be merged into master.

**Review by hexonaut — COMMENTED** [review]
Two minor issues. This is a good first pass translating the previous version. I'd like to do another PR after where the test logic is merged into a single base test contract and the specifics of the domain are dealt with by inheriting contracts. Similar to how this is done: https://github.com/makerdao/dss-bridge/tree/v1/src/tests/domains

Would also like to see the testing expanded to cover more of the features, but I think ordering can be so as to merge the test contracts first then adding more tests after.

File: .gitmodules
Comment (barrutko) [reply .gitmodules]: Updated

**Comment by barrutko** [comment]
Redone the structure and working on tests. Still some more to go.

**Review by hexonaut — CHANGES_REQUESTED** [review]
The `xchain-helpers` appears to be referencing a non-existent commit. Not sure what's going on here.

**Comment by barrutko** [comment]
> The `xchain-helpers` appears to be referencing a non-existent commit. Not sure what's going on here.

Should be fine now.

**Review by hexonaut — APPROVED** [review]
LGTM

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: What is this address? If you need a valid, random address it's better to use `makeAddr("guardian")` or similar.

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Typo, 3 "s" in Crosss

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Should not use assume when trying to convert a wide range into a small one as this `assume` will do trial and error to find valid values which is very wasteful. You should instead `bound()` function.

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Same here.

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Same here.

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Again I would use `makeAddr("notGuardian")`.

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: You are already on the L2 fork.

File: test/mocks/ReconfigurationPayload.sol
Comment (hexonaut) [inline test/mocks/ReconfigurationPayload.sol]: Would pass in a `makeAddr("newGuardian")` in the constructor.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Sorry I jumped the gun a bit with my last review. I didn't realize you merged 3 tasks into this PR.

Overall looks good, just some best practises to follow.

File: test/mocks/ReconfigurationPayload.sol
Comment (barrutko) [reply test/mocks/ReconfigurationPayload.sol]: I can't really do that. It's gonna be in the payload storage and the delegate call from executor is gonna run on the executor's storage.

Comment (barrutko) [reply test/mocks/ReconfigurationPayload.sol]: For now changed it to something more obviously meaningless.

Comment (barrutko) [reply test/mocks/ReconfigurationPayload.sol]: Ok, immutable is the thing 😅 Fixed.

**Review by hexonaut — DISMISSED** [review]
LGTM

File: test/CrosschainTestBase.sol
Comment (hexonaut) [inline test/CrosschainTestBase.sol]: Tab alignment.

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: test/CrosschainTestBase.sol
Comment (barrutko) [reply test/CrosschainTestBase.sol]: Fixed

**Review by hexonaut — DISMISSED** [review]

File: test/ArbitrumCrosschainTest.t.sol
Comment (barrutko) [inline test/ArbitrumCrosschainTest.t.sol]:
```suggestion
        payloadDataLength = bound(
```

File: test/ArbitrumCrosschainTest.t.sol
Comment (barrutko) [inline test/ArbitrumCrosschainTest.t.sol]:
```suggestion
    function testFuzz_arbitrumGasCalculations(uint256 paylodDataLength) public {
```

File: test/CrosschainTestBase.sol
Comment (barrutko) [inline test/CrosschainTestBase.sol]:
```suggestion
    function testFuzz_actionExecutionFailsBeforeTimelock(uint delay) public {
```

**Review by lucas-manuel — APPROVED** [review]
LGTM for a start, will build on this and add more testing as needed

**Review by hexonaut — APPROVED** [review]

**Pull request merged**
